docs/DeveloperGuide.md

```diff
@@ -1,38 +1,536 @@
+exception will be thrown and handled by printing an error message to the CLI.
+
+## Command Components
+
+### Help Command - Display Commands
+
+When executed, this command will execute a standard Final String containing all 
+the commands available for the user to use, as well as the format they are meant
+to be written in. 
+
+This same String is also displayed when an invalid command is used.
+
+### Short Command - Add Short Answer Question to the Quiz
+
+One of the supported question formats is the Short Answer Question, in which a question
+can be answered with a phrase of a few words, which will be matched to a corresponding answer
+of sufficient similarity.
+
+It is parsed as: `short [question]/[answer]/[module]/[difficulty]`
+1. `[question]` is the question, phrased as asking the user (i.e. What is 2 + 2? )
+2. `[answer]` is the answer or possible answer(s) that the user must
+   input to count as correct. It is case-insensitive. (i.e. 4, four)
+3. `[module]` is the module that the question belongs in (i.e. CS2113)
+4. `[difficulty]` is the difficulty of the question for sorting later (i.e. Hard)
+
+*Condensed Class Diagram - Does not contain all attributes & methods
+![](./UML/AddShortCommand.jpg)
+
+Thereafter, the command is returned to the Quizhub component and executed,
+to add a Question object to the corresponding QuestionList object
+using the `addToQuestionList` method. This method will analyse the arguments above
+and add the Question to the appropriate list.
+
+![](./UML/Commands/Initial State.png)
+![](./UML/Commands/commandAddState.png)
+
+### List Command - Show all Questions with Index
+
+Lists all the questions in the current QuestionList. 
+
+When executed, the command will invoke the QuestionList.printQuestionList method. 
+This method first checks whether the list is empty and prints and error message if empty, 
+else it will invoke the QuestionList.printQuestion method on each Question object, 
+with the asList parameter set as true. The QuestionList.printQuestion method will then
+print each question with a given index, and indicate the question's type and completion status
+(obtained through Question.getQuestionType() and Question.questionIsDone() methods)
+
+### Delete Command - Delete a Question
+
+Deletes a question by its index.
+
+The Command first invokes the QuestionList.viewQuestionByIndex method to determine if 
+the index refers to a valid question, failing which it will return an error message.
+
+Otherwise, it will then invoke the QuestionList.deleteQuestionByIndex method to remove
+the question from the QuestionList. 
+
+![](./UML/Commands/commandDeleteQuestion.png)
+
+Afterward, the storage is updated to reflect the newest QuestionList.
+
+### Find Command - Look for a matching question
+
+Quizhub supports searching for specific questions, by searching matching keywords
+in the format
+
+`find /[description]` OR `find /[module]`
+i.e. `find /water buffalo`, `find /CS2113`
```

Comment:
Perhaps the examples can be on a new line? It appears to be on the same line in the website. Maybe the format could be standardized for all the different commands.